libnetwork/driverapi/driverapi.go

```diff
@@ -156,12 +156,16 @@ type JoinInfo interface {
 	AddTableEntry(tableName string, key string, value []byte) error
 }
 
+// Registerer provides a way for network drivers to be dynamically registered.
+type Registerer interface {
+	RegisterDriver(name string, driver Driver, capability Capability) error
+}
+
 // DriverCallback provides a Callback interface for Drivers into LibNetwork
 type DriverCallback interface {
+	Registerer
```

Comment:
lol, this type reminds of https://www.youtube.com/watch?v=6kZBJs527-k
"The Rural Juror"